kotlin-sqlobject/src/test/kotlin/org/jdbi/v3/sqlobject/kotlin/CoroutineSqlObjectTest.kt

```diff
@@ -0,0 +1,131 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject.kotlin
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.jdbi.v3.core.Handle
+import org.jdbi.v3.core.Jdbi
+import org.jdbi.v3.core.kotlin.KotlinPlugin
+import org.jdbi.v3.core.kotlin.withExtensionUnchecked
+import org.jdbi.v3.sqlobject.SqlObject
+import org.jdbi.v3.sqlobject.SqlObjectPlugin
+import org.jdbi.v3.sqlobject.customizer.Bind
+import org.jdbi.v3.sqlobject.statement.SqlQuery
+import org.jdbi.v3.sqlobject.statement.SqlUpdate
+import org.jdbi.v3.testing.junit5.JdbiExtension
+import org.jdbi.v3.testing.junit5.internal.TestingInitializers
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+
+class CoroutineSqlObjectTest {
+
+    @RegisterExtension
+    @JvmField
+    val h2Extension: JdbiExtension = JdbiExtension.h2()
+        .withPlugin(KotlinPlugin(enableCoroutineSupport = true))
+        .withPlugin(SqlObjectPlugin())
+        .withInitializer(TestingInitializers.something())
+
+    private lateinit var jdbi: Jdbi
+
+    @BeforeEach
+    fun setup() {
+        jdbi = h2Extension.jdbi
+    }
+
+    data class Something(
+        val id: Int,
+        val name: String
+    )
+
+    @RegisterKotlinMapper(Something::class)
+    interface SomethingDao : SqlObject {
+        @SqlUpdate("insert into something (id, name) values (:id, :name)")
+        fun insert(@BindKotlin something: Something)
+
+        @SqlQuery("select id, name from something where id=:id")
+        fun findById(@Bind("id") id: Int): Something
+
+        @SqlQuery("select id, name from something")
+        fun list(): List<Something>
+    }
+
+    @Test
+    fun testTransactionIODispatcher() {
+        jdbi.withExtensionUnchecked(SomethingDao::class) { dao ->
+            val origHandle = dao.handle
+
+            val handle1: Handle?
+            val handle2: Handle?
+            val handle3: Handle?
+
+            runBlocking(Dispatchers.IO) {
+                coroutineScope {
+                    // all Jdbi operations will map onto the attached handle in this coroutine context
+                    handle1 = insertSomething(dao)
+
+                    handle2 = selectSomething(dao, 1, "first name")
+                    delay(1000L)
```

Comment:
1s is a fair amount of time to wait. Is it possible to write these tests with latches instead of timed sleep?

Reply:
I like the latch idea. I will see if I can change this tonight.